systemd: Port System overview to ct-form-layout

martinpitt opened the pull request:
As we don't have IDs for containing table rows any more, adjust the code
to directly interact with the action element, and possibly its label
(`.prev()`).

Similarly to commit 1570bf3624d, prevent the shutdown/restart button
click event from falling through to the parent form handler, to avoid
immediately closing the dialog that is about to open.

Adjust CSS selectors check-realms for the dropped intermediate
`button-location` span. This also still works for the old HTML on
rhel-7-6-distropkg, as there is no other span than the tooltip inside
`#system-info-domain`.

Fixes #9773

 - [x] Fix misplaced tooltip for hardware info
 - [x] Some form-layout fixes: PR #10974

martinpitt commented:
First iteration that reasonably fixes the functionality. I added the initial list of outstanding regressions to the description. @garrett, would you be able to help me with the first issue? That smells like CSS, but I didn't find it right away.

I'll get a complete test run from the bots and then work on adjusting the tests.  Not sure about the mis-placed tooltip, that might be a possible fix in ct-form-layout (correctly handling tooltips), or just a bug in the HTML structure; I'll check that.

martinpitt commented:
Overview with current master:
![overview-before](https://user-images.githubusercontent.com/200109/51118160-fb3a7880-180f-11e9-8800-4ee484da6f85.png)

Overview from this PR:
![overview-after](https://user-images.githubusercontent.com/200109/51118172-01c8f000-1810-11e9-8723-99e8447e33bf.png)

martinpitt commented:
@garrett: I put back the col-md-4 wrapper, so the layout change is once again independent.

martinpitt commented:
@garrett: That leaves the difference in padding. Do you think we should put back some extra one, like before, or is it fine as-is?

garrett commented:
I think it's fine as-is.

It was inconsistent with everything else before, and the additional padding was problematic on mobile.

That said, I think the form deconstruction is problematic on mobile now (in this PR), so I think I need to modify `ct-form-layout` — possibly adding another class or changing the breakpoint a bit.

The breakpoint was previously made based on being in dialogs, so perhaps I should keep it, if it is within a dialog, and change it for when form layouts are not in a dialog.

garrett commented:
As of the relayout using grid, machine ID is no longer truncated. Previously, the space provided by Bootstrap was either too little or too much, so we went with too little (and too much margins around everything) and truncated.

I'll fix mobile, see how it looks, and we should probably remove the machine ID tooltip afterward (as long as it isn't needed).

martinpitt commented:
I incorporated https://github.com/martinpitt/cockpit/pull/6 : squashed one commit, and extended the system ID one by dropping the now redundant tooltip. I checked this at various widths/heights and zoom levels, and it looks great now. Thanks!

I also pushed this on top of #10974, thus marking as blocked.

garrett commented:
@mvollmer: Yep! Intentional. Thanks for asking about it, BTW! I'm hoping it still works fine in German. :wink: (Haven't tested... yet?)

garrett commented:
Here's German. It's just a little more snug (thanks to "Secure Shell-Schlüssel"), but still works:

![sunny_9090_system iphone 5_se 1](https://user-images.githubusercontent.com/10246/51239768-7f107400-197a-11e9-8926-117dee3299e3.png)

Remember: This is basically the extreme low-res we support on phones, and I still think this is OK.

You can see how important the wrapping is. This is why I reworked the grid-layout yesterday to better support wrapping in cases like these. (For machine ID & OS in this example.)

martinpitt commented:
> Shouldn't the page also move the labels into their own rows?

FTR: Indeed I tested that with my original PR, and it did work. Now, with the form-layout rework it doesn't any more indeed. But it seems this was intended, and the rationale makes sense. Thanks!
